From 4749cc8fe4ffd83010db38bd6d943723c5e0c22d Mon Sep 17 00:00:00 2001
From: Icenowy Zheng <icenowy@aosc.io>
Date: Fri, 27 Jul 2018 17:34:35 +0800
Subject: [PATCH 286/329] nvmem: sunxi-sid: fix endian

The data in the SID is stored as little endian. However, when accessing
the SID via sunxi-sid driver, it converts little endian to big endian at
4 byte border. This makes addressing sub-4-byte items in the SID wrong.

Fix the endian by read out the SID 4-byte words as little endian, not
big endian.

Signed-off-by: Icenowy Zheng <icenowy@aosc.io>
---
 drivers/nvmem/sunxi_sid.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/nvmem/sunxi_sid.c b/drivers/nvmem/sunxi_sid.c
index fa58c3574afa..af30d62b1426 100644
--- a/drivers/nvmem/sunxi_sid.c
+++ b/drivers/nvmem/sunxi_sid.c
@@ -63,7 +63,7 @@ static u8 sunxi_sid_read_byte(const struct sunxi_sid *sid,
 {
 	u32 sid_key;
 
-	sid_key = ioread32be(sid->base + round_down(offset, 4));
+	sid_key = readl(sid->base + round_down(offset, 4));
 	sid_key >>= (offset % 4) * 8;
 
 	return sid_key; /* Only return the last byte */
-- 
2.18.0

